### PR TITLE
docs: update docs for the AWS X-Ray tracer

### PR DESCRIPTION
**Pull request by suniltheta**
Commit Message: update envoy docs to contain information related to AWS X-Ray extension and update codeowners for the same extension

Additional Description: Changes and their effect on envoy documentation:
1.  File: docs/root/configuration/http/http_conn_man/headers.rst: To include info about x-amzn-trace-id header consumed by x-ray extension.
[before](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers.html) and  [after](http://envoyxraydocs.s3-website-us-east-1.amazonaws.com/docs/configuration/http/http_conn_man/headers.html?highlight=amzn)

2. File: docs/root/intro/arch_overview/observability/tracing.rst: To include trace context propagation for AWS X-Ray
[before](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/observability/tracing.html) and [after](http://envoyxraydocs.s3-website-us-east-1.amazonaws.com/docs/intro/arch_overview/observability/tracing.html?highlight=X-Ray)

3. File: source/extensions/tracers/xray/BUILD: To remove `work-in-progress` Note on X-Ray API, as this extension is available.
[before](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/trace/v3/xray.proto.html?highlight=work-in-progress) and [after](http://envoyxraydocs.s3-website-us-east-1.amazonaws.com/docs/api-v3/config/trace/v3/xray.proto.html)

Risk Level: Low
Testing: N/A
Docs Changes: The PR updates the envoy documentation to reference X-Ray as a tracing extension that is supported by envoy and also gives reference towards the specific http header that is consumed by this extension.
Release Notes: N/A
Platform Specific Features: N/A

Signed-off-by: Sunil Narasimhamurthy <sunnrs@amazon.com>

### CODE REVIEW

File: bazel/repository_locations.bzl
Comment (lavignes) [inline bazel/repository_locations.bzl]: This is not necessary. these repository locations are for code dependencies that envoy needs to build and run.

**Review by lavignes — CHANGES_REQUESTED** [review]

**Comment by lavignes** [comment]
Meta comment. Your commit message starts with `Commit Message:`. That needn't be on there.

Also to make things easier on you, you should not merge this into your `main` branch :) if you update your main branch then it will make it more difficult to keep your fork in sync with `envoy/main`

**Comment by lavignes** [comment]
While we're at it can we also update the owner for this extension:

i.e. https://github.com/envoyproxy/envoy/blob/main/CODEOWNERS

You should replace `@marcomagdy` with `@abaptiste`.

Reason being that Marco no longer works for App Mesh. :)

File: bazel/repository_locations.bzl
Comment (suniltheta) [reply bazel/repository_locations.bzl]: I thought this section refers to possible external dependancies that are needed for envoy to work. I will remove this hunk entirely then.

**Comment by suniltheta** [comment]
> Meta comment. Your commit message starts with `Commit Message:`. That needn't be on there.

I will take care of this. Will follow the PULL_REQUESTS.md guide which Alvin pointed to throughly.

> Also to make things easier on you, you should not merge this into your `main` branch :) if you update your main branch then it will make it more difficult to keep your fork in sync with `envoy/main`

Ok then I will not merge this PR. But I will raise a new PR from **suniltheta/envoy-update_xray_documentation** branch towards **envoyproxy/envoy-main** branch. I will do this after getting approval from lavignes & abaptiste.

> While we're at it can we also update the owner for this extension:

I will also take care of updating OWNERS section for xray.

File: bazel/repository_locations.bzl
Comment (suniltheta) [reply bazel/repository_locations.bzl]: Fixed in the latest commit

**Comment by suniltheta** [comment]
Updated the CODEOWNERS for xray extension in the latest commit.

**Comment by lavignes** [comment]
The links to rendered docs are not necessary. They can also be rendered/viewed anyway as part of Envoy's CI when you make a PR

**Comment by abaptiste** [comment]
There's an "aookok " file that is empty and part of this commit  Can we make sure that only relevant files are included?

**Comment by abaptiste** [comment]
I like that you included the newly rendered docs.  As @lavignes mentioned, this isn't necessary for the Envoy PR, but I'm happy to see it here.

**Comment by suniltheta** [comment]
> There's an "aookok " file that is empty and part of this commit Can we make sure that only relevant files are included?

Alvin, yes I will get rid of this file. I was testing out **git push hook** which was not initially working on my ec2 local. There were some missing softwares that I had to install to make it work. Such as (aspell, buildozer, buildifier, go1.15.0+). This file is side effect of my testing to make sure the hooks are called without any issue. Now that this issue is resolved, in my fresh PR against upstream envoy I will not include this empty file.

**Pull request closed without merging**
